1 Niagara 4 Overview Exercise - BFile/myFile-rt/src/com/tridiumuniversity/myFile/BFile.java

```diff
@@ -0,0 +1,213 @@
+package com.tridiumuniversity.myFile;
+
+import javax.baja.nre.annotations.NiagaraAction;
+import javax.baja.nre.annotations.NiagaraProperty;
+import javax.baja.nre.annotations.NiagaraType;
+import javax.baja.sys.*;
+
+@NiagaraType
+
+@NiagaraProperty(
+        name = "fileName",
+        type = "String",
+        defaultValue = ""
+)
+@NiagaraProperty(
+        name = "path",
+        type = "String",
+        defaultValue = ""
+)
+@NiagaraProperty(
+        name = "size",
+        type = "int",
+        defaultValue = "0"
+)
+@NiagaraProperty(
+        name = "createdDate",
+        type = "BAbsTime",
+        defaultValue = "BAbsTime.now()"
+)
+@NiagaraProperty(
+        name = "modifiedDate",
+        type = "BAbsTime",
+        defaultValue = "BAbsTime.now()"
+)
+@NiagaraAction(
+        name = "touch",
+        flags = Flags.ASYNC
```

Comment:
By adding the `ASYNC` flag here, you're offloading the touch and print Actions to the control engine thread. That's usually not a good idea: either the Action is not computationally expensive, in which case it doesn't need to be offloaded to another thread at all, or the Action is computationally expensive enough to justify being offloaded to another thread, in which case you probably shouldn't put it on the engine thread.

You're doing so little in `doTouch` and `doPrint` that I would argue these Actions fall into the first category of not needing to be offloaded to another thread.

Reply:
Thank you for the clarification!